querybook/webapp/components/DataDocRightSidebar/DataDocRunAllButtonConfirm.tsx

```diff
@@ -27,6 +30,20 @@ export const DataDocRunAllButtonConfirm: React.FunctionComponent<IProps> = ({
         [onNotificationChange, setNotification]
     );
 
+    let queryCells = useQueryCells(docId);
+    if (index !== undefined) {
+        queryCells = queryCells.slice(index);
+    }
```

Comment:
is this the index of the query cell in the datadoc, including text cells?  e.g. what if there is a text cell before the starting query cell, is the slicing still correct?

same for the `start_index` in the above .py file, please double confirm